tests/internal/tls/validFor_test.go

```diff
@@ -0,0 +1,162 @@
+	clientCert  *tls.Certificate
+}
+
+func (m *mockIdentity) Cert() *tls.Certificate         { return m.clientCert }
+func (m *mockIdentity) ServerCert() []*tls.Certificate { return m.serverCerts }
+func (m *mockIdentity) CA() *x509.CertPool             { return nil }
+func (m *mockIdentity) CaPool() *identity.CaPool       { return nil }
+func (m *mockIdentity) ServerTLSConfig() *tls.Config   { return nil }
+func (m *mockIdentity) ClientTLSConfig() *tls.Config   { return nil }
+func (m *mockIdentity) Reload() error                  { return nil }
+func (m *mockIdentity) WatchFiles() error              { return nil }
+func (m *mockIdentity) StopWatchingFiles()             {}
+func (m *mockIdentity) SetCert(pem string) error       { return nil }
+func (m *mockIdentity) SetServerCert(pem string) error { return nil }
+func (m *mockIdentity) GetConfig() *identity.Config    { return nil }
+func (m *mockIdentity) ValidFor(address string) bool   { return true }
+
+const (
+	validDNS   = "example.com"
+	invalidDNS = "invalid.com"
+	validIP4   = "192.168.1.1"
+	invalidIP4 = "10.0.0.1"
+	validIP6   = "::1"
+	invalidIP6 = "fe80::1"
+	validPort  = "443"
+)
+
+// Helper to create a mock identity with certs
+func createmockIdentity(dnsNames []string, ipAddresses []string) *identity.TokenId {
+	leaf := &x509.Certificate{}
+	for _, dns := range dnsNames {
+		leaf.DNSNames = append(leaf.DNSNames, dns)
+	}
+	for _, ip := range ipAddresses {
+		leaf.IPAddresses = append(leaf.IPAddresses, net.ParseIP(ip))
+	}
+
+	tlsCert := &tls.Certificate{Leaf: leaf}
+	mi := &mockIdentity{
+		serverCerts: []*tls.Certificate{tlsCert},
+		clientCert:  tlsCert,
+	}
+	id := &identity.TokenId{
+		Identity: mi,
+		Token:    "",
+		Data:     nil,
+	}
+	return id
+}
+
+func TestValidFor_ValidHostname(t *testing.T) {
+	id := createmockIdentity([]string{validDNS}, []string{})
+
+	err := ziti_tls.ValidFor(id, validDNS+":"+validPort)
+	if err != nil {
+		t.Errorf("Expected valid hostname, got error: %v", err)
+	}
+}
+
+func TestValidFor_InvalidHostname(t *testing.T) {
+	id := createmockIdentity([]string{validDNS}, []string{})
+
+	err := ziti_tls.ValidFor(id, invalidDNS+":"+validPort)
+	if err == nil {
+		t.Errorf("Expected error for invalid hostname, got nil")
+	}
+	assert.Equal(t, "identity is not valid for provided host: ["+invalidDNS+"]. is valid for: ["+validDNS+"]", err.Error())
+}
+
+func TestValidFor_ValidIPv4(t *testing.T) {
+	id := createmockIdentity([]string{}, []string{validIP4})
+
+	err := ziti_tls.ValidFor(id, validIP4+":"+validPort)
+	if err != nil {
+		t.Errorf("Expected valid IP, got error: %v", err)
+	}
+}
+
+func TestValidFor_InvalidIPv4(t *testing.T) {
+	id := createmockIdentity([]string{}, []string{validIP4})
+
+	err := ziti_tls.ValidFor(id, invalidIP4+":"+validPort)
+	if err == nil {
+		t.Errorf("Expected error for invalid IP, got nil")
+	}
+	assert.Equal(t, "identity is not valid for provided host: ["+invalidIP4+"]. is valid for: ["+validIP4+"]", err.Error())
```

Comment:
We have the `require` library in the project to make test asserts more compact and expressive.

```
req := require.New(t)
...
req.NoError(err)
req.NoError(err, "expected valid hostname got error, %v", err))
req.Equal(1,2)
req.Equal(1,value, "value was expected to be 1, got %v", value)
```

Making you aware of it because I don't know if you have written tests in this project before.

Reply:
i'll adapt the tests over there